instructions/instructions_impl.cpp

```diff
@@ -94,15 +130,20 @@ void Mv::exec(State &state) {
   state.registers[dist] = state.registers[source];
 }
 
-void Mv::fill_args(vector<string> args) {
-  // converted to types: Register, Register
-  int args_amount = 2;
+
+Or::Or(vector<string> args) {
+  // converted to types: Register, Register, Register
+  int args_amount = 3;
   if (args.size() != args_amount) {
-    throw ParserException(
-        Parser::exception_message("mv", args_amount, args.size()));
+    throw ParserException("or", args_amount, args.size());
   }
-  dist = Parser::get_register(args[0]);
-  source = Parser::get_register(args[1]);
+  Register dist_ = Parser::get_register(args[0]);
+  Register source1_ = Parser::get_register(args[1]);
+  Register source2_ = Parser::get_register(args[2]);
+
+  dist = dist_;
+  source1 = source1_;
+  source2 = source2_;
```

Comment:
Why use this buffers

Reply:
To avoid partial filling of fields, these functions (like get_register) can throw exceptions